changelog/dmd.intel-cet-ibt-protection.dd

```diff
@@ -0,0 +1,22 @@
+Added support for Intel CET (Control-flow Enforcement Technology) IBT (Indirect Branch Tracking) protection
+
+CET is a technology that is useful for preventing an attacker from redirecting a program's control flow,
+specifically IBT prevents an attacker from causing an indirect branch to go to an unintended place.
+
+Intel IBT expects the compiler to emit special instructions (`endbr32` and `endbr64`) which in older processors
+that do not support IBT are equivalent to `nop` instructions, consequently a program compiled with active IBT
+will be compatible on any x86 processor and the protection will be opportunistically active on supported processors.
+
+To enable Intel IBT protection in DMD you need to pass the `-fIBT` flag to the compiler, consequently the compiler
+will manage the emission of instructions for IBT by itself.
+Be careful when using inline assembly, the compiler will not automatically handle IBT inside an inline assembly.
+
+To find out within a D program whether IBT has been activated or not use the traits getTargetInfo as follows:
+
+---
+// IBT active
+static assert(__traits(getTargetInfo, "CET") == 1);  // CET == 1 if IBT is active
```

Comment:
I'm reluctant to call this "CET" when the switch is "IBT". CET includes shadow stack support, which DMD does not yet support. When it does support CET, I'll add a -fCET and only then should getTargetInfo say it supports CET.

Reply:
CET includes both IBT and shadow stack, LDC use `__traits(getTargetInfo, "CET") ` as follows:

- Disabled: `__traits(getTargetInfo, "CET") == 0`
- IBT (--fcf-protection=branch in LDC): `__traits(getTargetInfo, "CET") == 1`
- Shadow stack (--fcf-protection=return in LDC): `__traits(getTargetInfo, "CET") == 2`
- IBT + Shadow stack (--fcf-protection=full in LDC): `__traits(getTargetInfo, "CET") == 3`

This behavior is equivalent to `__CET__` of C compilers

Reply:
ok then